src/main/java/com/avail/descriptor/StringDescriptor.java

```diff
@@ -108,6 +109,51 @@ protected final void o_WriteTo (final AvailObject object, final JSONWriter write
 		writer.write(object.asNativeString());
 	}
 
+	/**
+	 * Convert the specified Java {@link String} to an Avail {@link A_String},
+	 * but keeping any Java surrogate pairs as two distinct values in the Avail
+	 * string.  Note that such a string is semantically different from what
+	 * would be produced by {@link #stringFrom(String)}, and isn't even
+	 * necessarily the same length.  This operation is intended for
+	 * compatibility with Java (and JavaScript) strings.
+	 *
+	 * <p>NB: The {@linkplain AbstractDescriptor descriptor} type of the actual
+	 * instance returned varies with the contents of the Java {@code String}. If
+	 * the Java {@code String} contains only Latin-1 characters, then the
+	 * descriptor will be {@link ByteStringDescriptor}; otherwise it will be
+	 * {@link TwoByteStringDescriptor}.</p>
+	 *
+	 * @param aNativeString
+	 *        A Java {@link String}.
+	 * @return An Avail {@code StringDescriptor string} having the same length,
+	 *         but with surrogate pairs (D800-DBFF and DC00-DFFF) preserved in
+	 *         the Avail string.
+	 */
+	public static A_String stringWithSurrogatesFrom (final String aNativeString)
```

Comment:
I'm surprised you don't recognize this.  I coded it for Rich's Anvil work, so that the strings Avail manipulates and the strings JavaScript (or Java) manipulates can have the same length – but only when you choose to use UCS-16 like this.